docs/BindingWithGrammar.md

```diff
@@ -0,0 +1,131 @@
+# Binding with Grammar
+
+[vscode-xml](https://github.com/redhat-developer/vscode-xml) provides support for automatic XML schema/grammar binding through generation and/or using an existing `.xsd/.dtd` file.
+
+[Validation with XSD grammar](Validation.md#validation-with-xsd-grammar) and [Validation with DTD grammar](Validation.md#validation-with-dtd-grammar) provide more information on manual binding of an XML document.
+
+In the following sections, we would like to bind the XML document `foo.xml`:
+
+```xml
+<foo>
+  <bar />
+</foo>
+```
+
+with an associated grammar file.
+
+## Binding with Existing Grammar
+
+Consider an XSD document, `foo.xsd` (in the same directory as `foo.xml`), defined as follows :
+
+```xsd
+<?xml version="1.0" encoding="UTF-8"?>
+<xs:schema xmlns:xs="http://www.w3.org/2001/XMLSchema">
+  <xs:element name="foo">
+    <xs:complexType>
+      <xs:sequence>
+        <xs:element name="bar" />
+      </xs:sequence>
+    </xs:complexType>
+  </xs:element>
+</xs:schema>
+```
+
+and a DTD document, `foo.dtd` (in the same directory as `foo.xml`), defined as follows:
+```dtd
+<!ELEMENT foo (bar)>
+<!ELEMENT bar EMPTY>
+```
+
+Either file can be bound using the [XML Binding Wizard](#the-xml-binding-wizard), which can be triggered by [Command](#command), [CodeLens](#codelens) or [Quick Fix](#quick-fix).
+
```

Comment:
I think you should add a sample demo with CodeLens here to see in action the process and see that you can select xsd file.